src/shogun/mathematics/linalg/internal/implementation/Scale.h

```diff
@@ -66,6 +66,20 @@ struct scale
 	static void compute(Matrix A, Matrix B, Matrix C, T alpha, T beta);
 };
 
+/** Specialization of scale for the Native backend */
+template<> template<class Matrix>
+struct scale<Backend::NATIVE, Matrix>
+{
+		typedef typename Matrix::Scalar T;
+		
+		/** Performs the operation B = alpha*A */
+		static void compute(SGVector<T> A, SGVector<T> B, T alpha)
```

Comment:
That's fine. The main thing is that these specializations handle both matrices and vector. For our native implementation we can add any helper method we need to achieve that purpose. This is practically a different class altogether.